core/dynamo.c

```diff
@@ -2571,7 +2571,7 @@ dynamo_thread_exit_common(dcontext_t *dcontext, thread_id_t id,
 #endif
 
 #ifdef DEBUG
-    if (dcontext->logfile != INVALID_FILE) {
+    if (dcontext->logfile != INVALID_FILE && dcontext->logfile != STDERR) {
```

Comment:
The same check should be done when closing main_logfile, right?

Reply:
In this file there is already a check to verify `main_logfile != STDERR` before closing it. Is there elsewhere in the codebase that this needs to happen? I looked at all calls to `close_log_file` in the tree and I couldn't find any relevant ones.

Reply:
Oh, ok, did not see that the main logfile is already checked.